### PR TITLE
TypeScript, ESM, and more

### PR DESCRIPTION
**Pull request by ackledotdev**
## Major changes
- Use TypeScript
- Use ESM over CJS
- Bump minor version
- Use JSDoc with theme instead of Docma

## Minor changes
- Use ESLint config
- Add NPM ignore file
- Update formatter
- Remove old dependencies

### CODE REVIEW

**Comment by ackledotdev** [comment]
fixes #35

**Comment by ackledotdev** [comment]
fixes #30

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+	"trailingComma": "none",
+	"tabWidth": 2,
+	"semi": true,
+	"singleQuote": true,
+	"useTabs": true,
```
Comment (khalby786) [inline .prettierrc]: Can we set this to `false`? I think it'd be more standard to use spaces here.

File: package.json [hunk 2]
```diff
@@ -1,52 +1,54 @@
...
-    "url": "https://github.com/khalby786/jsoning"
-  },
-  "engines": {
-    "node": ">=0.12"
-  },
-  "files": [
-    "src"
-  ]
+	"name": "jsoning",
+	"version": "0.14.0",
```
Comment (khalby786) [inline package.json]: In light of un-deprecating this package with a new Typescript base, I think we ought to make this v1.0.0

**Review by khalby786 — CHANGES_REQUESTED** [review]

**Comment by khalby786** [comment]
Apart from the requested changes (which are cosmetic), LGTM! I'm afk right now, and I'll merge it when I get home after reviewing `index.ts` thoroughly because I've basically forgotten what my original code was - I need to familiarize myself with the code so that I can continue working on it!

**Comment by khalby786** [comment]
Apart from the requested changes (which are cosmetic), LGTM! I'm afk right now, and I'll merge it when I get home after reviewing `index.ts` thoroughly because I've basically forgotten what my original code was.

**Comment by khalby786** [comment]
This is great work btw! Thanks a ton for motivating me to bring back this package!

File: .prettierrc (hunk 1 above)
Comment (ackledotdev) [reply .prettierrc]: done!

File: package.json (hunk 2 above)
Comment (ackledotdev) [reply package.json]: also done!

**Review by khalby786 — APPROVED** [review]

**Pull request merged**